### DISCUSSION TITLE
Using this inside a webview (on IOS)

### DISCUSSION THREAD

**Discussion by DimaM-BeStreet** [add2cal/add-to-calendar-button#100] [Q&A] [answered]
Hey! Again - love what you did <3

Im trying to implement this button into a page opened by a webview on IOS (iphones)

After reading a bit about this i think there is no good way to do that without handling the whole download process in native code (way too messy for me right now)
BUT - am asking anyway, guess i'm feeling optimistic =]

Is there any way?
Maybe the click/tap on the button can automaticly open a new page in safari and ask the user to download the ics file?
(as an added feature, not by default)

**Comment by jekuer** [comment]
Unfortunately, this is not possible.
We already implemented a workaround for the Instagram in-app Browser.
Maybe we need to extend this to all webviews. **[Update: Will be included in the next release]**

At this workaround, whenever the user is in that situation, for ical only, we automatically copy the target url into the user's clipboard and show a message that he needs to open the main browser manually and paste the code there.

On Android, it is also helpful since ics data is sometimes blocked (not always as with iOS). However, if an ics file is given explicitly, we can skip the workaround on Android.

**Comment by DimaM-BeStreet** [comment, marked as answer]
I've sort of implemented my own thing and it seems to work nicely :)

What i did was create a very simple page that recieves parameters in the url and from them cretes an invitation to save to calendar (using your button)

This way i send the user to that page from my webview (it openes in the external default browser (safari by default))

Thank you again! hope this helps anyone with the webview problem! :)
Dima.

**Reply by jekuer** [reply]
If you want to, you could also check out the latest release, which offers some workaround by presenting a info note to the user while automatically copying the url to the clipboard. 

Not ideal, but the most generic solution, since some WebView apps also block the whole "open a link in safari" flow (e.g. Instagram).

If you can be sure your WebView page does not pop up there (e.g. because it is your app), your solution would be better!

**Reply by DimaM-BeStreet** [reply]
Great thank you! :)
For me this worked good! moving on to the next thing much faster thanks to you <3
